crates/sui-graphql-rpc/src/types/event.rs

```diff
@@ -217,7 +217,13 @@ impl Event {
                         }
 
                         if let Some(type_) = &filter.event_type {
-                            query = type_.apply(query, events::dsl::event_type);
+                            query = type_.apply(
+                                query,
+                                events::dsl::event_type,
```

Comment:
I somehow remembered that we will use event_type_package/module/name instead of event_type, why do we apply all of them here, what's the generated query like here?

Reply:
So for a type like `0x2::coin::Coin<0x2::sui::SUI>`, the `package`, `module` and `name` fields will be `0x2`, `coin` and `Coin`, respectively. Notice the lack of type parameters here. And the `type` field will be the fully qualified type `0x2::coin::Coin<0x2::sui::SUI>` with type parameters. We still need the `event_type` to do exact match when the full type is present in the filter.